MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/account/AccountRetrievalService.java

```diff
@@ -0,0 +1,163 @@
+package com.mapbox.mapboxandroiddemo.account;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+import android.util.Log;
+
+import com.mapbox.mapboxandroiddemo.MainActivity;
+import com.mapbox.mapboxandroiddemo.model.usermodel.UserResponse;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+/**
+ * Background service which retrieves Mapbox Account information
+ */
+
+public class AccountRetrievalService extends IntentService {
+
+  private static final String BASE_URL = "https://api.mapbox.com/api/";
+  private static final String ACCESS_TOKEN_URL = "https://api.mapbox.com/oauth/access_token";
+
+  //  TODO: Fill in CLIENT_SECRET and before running app. Never commit changes to Github with CLIENT_SECRET filled in!
+  private static final String CLIENT_SECRET = "";
+
+  private String clientId;
+  private String redirectUri;
+  private String username;
+
+  public AccountRetrievalService() {
+    super("AccountRetrievalService");
+  }
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+  }
+
+  @Override
+  protected void onHandleIntent(@Nullable Intent intent) {
+    if (intent != null) {
+      String authCode = intent.getStringExtra("AUTHCODE");
+      clientId = intent.getStringExtra("clientId");
+      redirectUri = intent.getStringExtra("redirectUri");
+      getAccessToken(authCode);
+    } else {
+      Log.d("AccountRetrievalService", "onHandleIntent: intent == null");
+    }
+  }
+
+  private void getAccessToken(String code) {
+    OkHttpClient client = new OkHttpClient();
+
+    Request request = new Request.Builder()
+      .addHeader("User-Agent", "Android Dev Preview")
+      .addHeader("Content-Type", "application/x-www-form-urlencoded")
+      .url(ACCESS_TOKEN_URL)
+      .post(RequestBody.create(MediaType.parse("application/x-www-form-urlencoded"),
+        "grant_type=authorization_code&client_id=" + clientId + "&client_secret=" + CLIENT_SECRET
+          + "&redirect_uri=" + redirectUri + "&code=" + code))
+      .build();
+
+    client.newCall(request).enqueue(new okhttp3.Callback() {
+      @Override
+      public void onFailure(okhttp3.Call call, IOException exception) {
+
+      }
+
+      @Override
+      public void onResponse(okhttp3.Call call, okhttp3.Response response) throws IOException {
+
+        String json = response.body().string();
+
+        JSONObject data = null;
+        try {
+          data = new JSONObject(json);
```

Comment:
Yea, I knew that my configuration was off. I guess the question is/was whether it's worth spending the time to fiddle with the configuration until it works or just leave the okhttp3 as is. I think I know your answer 😉